### DISCUSSION TITLE
이건 좀 심각한 버그인데...

### DISCUSSION THREAD

**Discussion by peterkangdev** [naverz/zepeto-studio-kor#406] [Issue Report (Legacy)]
아래 동영상을 보시면 (iPhone12, iOS 14.8.1 - Zepeto Package 는 현재시각 최신버전임, 제페토앱은 3.9.8)

- Create 버튼을 한번, 두번 클릭합니다. (전혀 작동하지 않음)
- 그런 후 Create 버튼을 3초간 누루고 있다가 떼면 작동합니다.

CUBE 가 만들어진 후 

- DeleteAll 버튼을 한번, 두번 클릭합니다.  (전혀 작동하지 않음)
- 그런 후 DeleteAll 버튼을 3초간 누루고 있다가 떼면 작동합니다.

이 상황이 심각한 이유는 IWP 에서 버튼을 작동할때 같은 현상이 반복됩니다. 
혹시 다른 분은 어떻게 해결하셨는지 모르겠지만, 심각한 상황인게 맞는것 같습니다.
해결이 되었으면 좋겠네요..

https://user-images.githubusercontent.com/30133892/154845224-8baf5afa-0b49-4aad-9ef9-8c6c3e10ce90.mov

**Comment by x4cf** [comment]
버튼은 즉시응답하고 있습니다. 혹시 코드를 볼 수 있을까요?

**Reply by peterkangdev** [reply]
제페토에서 공게한 UIEvent script 코드 입니다. 

이건 저만 그런게 아니고, 다른 분도 같은 이슈로 #369 문제제기 했습니다.

**Comment by peterkangdev** [comment]
자체적으로 해결했습니다.
iPhone 특성을 타는것 같습니다. iPhone12, iPhoneXS 에서 문제가 발생했습니다. 
잘되는 분들도 있고, 저 처럼 어려움을 겪는 분들도 계시겠지요.

IWP 에서 
일단 저와 같은 문제를 겪으시는 분들은 다음과 같이 해 보세요.

1. IWP Button 을 직접 사용하지 마시고, 버튼을 보이지 않게 투명하게 만듭니다. 
2. Image 를 IWP Button 처럼 만듭니다.
3. Script 를 만들어서 Image 를 터치하면 PointClick Event 를 발생시킵니다.
4. PointClick 이 일어나면 IWP Button 의 onClick.Invoke() 함수를 실행합니다.

다소 복잡하지만 위와 같이 하면 우회적으로 구현할 수 있습니다.

**Comment by cheolhui-lee** [comment]
안녕하세요.
중요한 이슈 말씀 주셔서 고맙습니다.
해당 내용 파악 중이고, 다음 3월 업데이트에선 개선 될 수 있도록 하겠습니다.
고맙습니다.
